utils/swift_build_support/swift_build_support/host_specific_configuration.py

```diff
@@ -186,6 +188,13 @@ def __init__(self, host_target, args):
                         "check-swift{}-optimize_none_with_implicit_dynamic-{}"
                         .format(subset_suffix, name))
 
+            # Only pull in these flags when cross-compiling with
+            # --cross-compile-hosts.
+            if deployment_target_name != args.host_target and \
+                host_target != args.host_target:
+                self.swift_flags = deployment_target.swift_flags(args)
+                self.cmake_options = deployment_target.cmake_options(args)
```

Comment:
Note that I didn't name these Python functions with `cross` or anything else related to cross-compilation like I had the bash functions in the first commit, so that if you ever want to use them for native compilation too, just remove these two checks above and go.